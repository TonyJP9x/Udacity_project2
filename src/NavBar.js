import React from "react";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { logout } from "./Slices/AuthSlice";

function NavBar(props) {
  const dispatch = useDispatch();
  const userInfo = useSelector(state => state.login.value)
  const handleOnClick = () =>  {
    dispatch(logout())
  }
  return (
    <div style={{marginBottom: '2px'}}>
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
          <div className="row" style={{width:'100%'}}>
            <div className="col-10" style={{display:'flex', paddingLeft:'30px'}}>
              <Link to='/home' className="navbar-brand" >
                Home
              </Link>
              <Link to='/leaderboard' className="navbar-brand" >
                LeaderBoard
              </Link>
              <Link to='/add' className="navbar-brand" >
                New
              </Link>
            </div>
            <div className="col-2" style={{display: 'flex'}} >
                <Link className="navbar-brand"> {userInfo.id}</Link>
                <img src={userInfo.avatarURL} style={{width:'40px', borderRadius:'50%', marginRight: '20px'}}/>
                <Link to='/' className="navbar-brand" onClick={handleOnClick}>Logout</Link>
            </div>
          </div>

      </nav>
    </div>
  );
}

export default NavBar;
