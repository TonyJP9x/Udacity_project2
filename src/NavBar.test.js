import NavBar from "./NavBar";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "./Store";
import { MemoryRouter } from "react-router-dom";

describe('navigation bar',() => {

    it("will have all expected field", () => {
        var component = render(
          <MemoryRouter>
            <Provider store={store}>
              <NavBar />
            </Provider>
          </MemoryRouter>
        );
        expect(component).toMatchSnapshot();
      });

  
})