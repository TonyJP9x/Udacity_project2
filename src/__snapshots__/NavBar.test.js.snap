// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation bar will have all expected field 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin-bottom: 2px;"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <div
            class="row"
            style="width: 100%;"
          >
            <div
              class="col-10"
              style="display: flex; padding-left: 30px;"
            >
              <a
                class="navbar-brand"
                href="/home"
              >
                Home
              </a>
              <a
                class="navbar-brand"
                href="/leaderboard"
              >
                LeaderBoard
              </a>
              <a
                class="navbar-brand"
                href="/add"
              >
                New
              </a>
            </div>
            <div
              class="col-2"
              style="display: flex;"
            >
              <a
                class="navbar-brand"
                href="/"
              >
                 
              </a>
              <img
                style="width: 40px; border-radius: 50%; margin-right: 20px;"
              />
              <a
                class="navbar-brand"
                href="/"
              >
                Logout
              </a>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin-bottom: 2px;"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          class="row"
          style="width: 100%;"
        >
          <div
            class="col-10"
            style="display: flex; padding-left: 30px;"
          >
            <a
              class="navbar-brand"
              href="/home"
            >
              Home
            </a>
            <a
              class="navbar-brand"
              href="/leaderboard"
            >
              LeaderBoard
            </a>
            <a
              class="navbar-brand"
              href="/add"
            >
              New
            </a>
          </div>
          <div
            class="col-2"
            style="display: flex;"
          >
            <a
              class="navbar-brand"
              href="/"
            >
               
            </a>
            <img
              style="width: 40px; border-radius: 50%; margin-right: 20px;"
            />
            <a
              class="navbar-brand"
              href="/"
            >
              Logout
            </a>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
